evm_trace/vmtrace.py

```diff
@@ -1,192 +1,186 @@
-    storage: dict[int, int]
-
-
-def to_trace_frames(
-    trace: VMTrace,
-    depth: int = 1,
-    address: str = "",
-    copy_memory: bool = True,
-) -> Iterator[VMTraceFrame]:
-    """
-    Replays a VMTrace and yields trace frames at each step of the execution.
-    Can be used as a much faster drop-in replacement for Geth-style traces.
-
-    Args:
-        trace (VMTrace): A decoded trace from a `trace_` rpc.
-        depth (int): A depth of the call being processed. automatically populated.
-        address (str): The address of the contract being executed. auto populated
-            except the root call.
-        copy_memory (bool): Whether to copy memory when returning trace frames.
-            Disable for a speedup when dealing with traces using a large amount of memory.
-            when disabled, `VMTraceFrame.memory` becomes `memoryview` instead of `bytes`, which
-            works like a pointer at the memory `bytearray`. this means you must process the
-            frames immediately, otherwise you risk memory value mutating further into execution.
-
-    Returns:
-        Iterator[VMTraceFrame]: An iterator of synthetic traces which can be used as a drop-in
-        replacement for Geth-style traces. also contains the address of the current contract
-        context.
-    """
-    memory = Memory()
-    stack = Stack()
-    storage: dict[int, int] = {}
-    call_address = ""
-    read_memory = memory.read_bytes if copy_memory else memory.read
-
-    for op in trace.ops:
-        if op.ex and op.ex.mem:
-            memory.extend(op.ex.mem.off, len(op.ex.mem.data))
-
-        # geth convention is to return after memory expansion, but before the operation is applied
-        yield VMTraceFrame(
-            address=address,
-            pc=op.pc,
-            op=op.op,
-            depth=depth,
-            stack=[to_int(val) for val in stack.values],
-            memory=read_memory(0, len(memory)),
-            storage=storage.copy(),
-        )
-
-        if op.op in ["CALL", "DELEGATECALL", "STATICCALL"]:
-            call_address = Address.__eth_pydantic_validate__(stack.values[-2])
-
-        if op.ex:
-            if op.ex.mem:
-                memory.write(op.ex.mem.off, len(op.ex.mem.data), op.ex.mem.data)
-
-            num_pop = POPCODES.get(op.op)
-            if num_pop:
-                stack.pop_any(num_pop)
-
-            for item in op.ex.push:
-                stack.push_bytes(item)
-
-            # erigon bug: https://github.com/ledgerwatch/erigon/pull/7970
-            if op.op == "PUSH0" and not op.ex.push:
-                stack.push_int(0)
-
-            if op.ex.store:
-                storage[op.ex.store.key] = op.ex.store.val
-
-        if op.sub:
-            yield from to_trace_frames(
-                op.sub, depth=depth + 1, address=call_address, copy_memory=copy_memory
+import sys
+
+# msgspec does not currently build on python 3.13 so hide it behind a version check
+if sys.version_info < (3, 13):
```

Comment:
**idea**: so one thing you could do in this situation to avoid having to indent everything is move all of the contents of the file to another file called `_vmrace.py` (leading underscore). Then, in this file import all those things and put them in __all__ iff this condition.